api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java

```diff
@@ -426,7 +448,10 @@ public void registerListener(Plugin plugin, Listener listener)
                     "Listener %s has registered using deprecated subscribe annotation! Please update to @EventHandler.", listener );
         }
         eventBus.register( listener );
```

Comment:
unregister is done in a synchronized context below but register isnt here, should be consistent.

Reply:
I originally figured that this was okay since EventBus maintains its own locking, but now I realise this could lead to inconsistent state.